### DISCUSSION TITLE
Changed behaviour in the way `persist_result` works

### DISCUSSION THREAD

**Discussion by SantanM** [PrefectHQ/prefect#16521] [Q&A] [answered]
In Prefect v2.19.9, for a flowrun/taskrun whenever the option `persist_result` is enabled, the table `artifact` has a record (that is of type `result`) corresponding to that flowrun/taskrun. However, in Prefect v3.0.11, the setting `persist_result=true` does not create record in `artifact` table for a flowrun/taskrun. Sample screenshots of the APIs with the difference
v2.19.9
```
curl 'http://localhost:4200/api/artifacts/filter' \
  -H 'Accept: application/json, text/plain, */*' \
  -H 'Content-Type: application/json' \
  --data-raw '{"artifacts":{"key":{},"type":{"any_":["result"]},"flow_run_id":{"any_":["aca67d43-12af-4f00-b0e5-16813539b126"]}}}'
```
![image](https://github.com/user-attachments/assets/2b5072d4-759c-41bb-b11e-31fd9beffb33)

v3.0.11
```
curl --location 'http://localhost:41120/api/artifacts/latest/filter' \
--data '{"artifacts":{"key":{},"type":{"any_":["result"]},"flow_run_id":{"any_":["862b42d4-d324-415a-9706-2a24bfb865ce"]}}}'
```
![image](https://github.com/user-attachments/assets/f056535e-6978-48ce-83d0-563e7641b1df)

**Comment by parkedwards** [comment, marked as answer]
hey @SantanM -- this is expected behavior.  In Prefect 2, the `persist_results` kwarg used to store the run result into configured storage, and create a reference to that storage location via an Artifact entry

In Prefect 3, we can still use the `persist_results` kwarg to do the same thing, but a resulting Artifact entry is no longer created (as it used to just hold a reference to the location, which is already information stored on the state object).  Depending on your goals, you can either:

(1) continue to use the `persist_results` kwarg and [configure the storage based on your needs](https://docs.prefect.io/v3/develop/results), or
(2) use one of the explicit `create_<type>_artifact()` [methods to store any result data as an Artifact](https://docs.prefect.io/v3/develop/artifacts#create-artifacts)
